lang/LangPrimSource/OSCData.cpp

```diff
@@ -925,6 +926,9 @@ void cleanup_OSC() {
     postfl("cleaning up OSC\n");
 
     stopAsioThread();
+    // destroy socket after ASIO thread has been joined!
+    // NOTE: don't rely on the global object destructor!!
```

Comment:
Maybe we can expand this a bit:

```
// NOTE: the socket must be destroyed *before* the IO service, so we cannot rely on the global object destructor because the order would be undefined.
```